tflog/options_test.go

```diff
@@ -0,0 +1,114 @@
+package tflog_test
+
+import (
+	"bytes"
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/terraform-plugin-log/internal/loggertest"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+func testSubsystemTraceHelper(ctx context.Context, message string) {
+	tflog.SubsystemTrace(ctx, testSubsystem, message)
+}
+
+func TestWithAdditionalLocationOffset(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		additionalLocationOffset int
+		logImpl                  func(context.Context)
+		expectedOutput           []map[string]interface{}
+	}{
+		"0-no-helper": {
+			additionalLocationOffset: 0,
+			logImpl: func(ctx context.Context) {
+				tflog.SubsystemTrace(ctx, testSubsystem, "test message")
+			},
+			expectedOutput: []map[string]interface{}{
+				{
+					"@caller":  "/tflog/options_test.go:30",
```

Comment:
I think it would be beneficial for future contributors to know how the integer/line is determined.

I can see someone coming here, changing something like an import or adding a test case above or whatever, and seeing that suddenly all those tests are failing.

A note above the line that warns that the `:line` part of that string is dependent on the very content of the file is in, it can save someone a bit of head scratching.

Reply:
Updated in https://github.com/hashicorp/terraform-plugin-log/pull/36/commits/32c00605582437ed9935c17f49fd3c1e63e34dd3